exec/java-exec/src/main/java/org/apache/drill/exec/physical/base/AbstractGroupScanWithMetadata.java

```diff
@@ -453,6 +458,14 @@ public boolean supportsLimitPushdown() {
   public GroupScan applyLimit(int maxRecords) {
     maxRecords = Math.max(maxRecords, 1); // Make sure it request at least 1 row -> 1 file.
     GroupScanWithMetadataFilterer<?> prunedMetadata = getFilterer();
+
+    // Return the group scan with the limit pushed down
+    if (this.maxRecords != maxRecords) {
```

Comment:
Placing this block here will also break files pruning logic. If the condition is true, pruning logic wouldn't be applied. Please move it to the places, where null is returned.

Reply:
@vvysotskyi 
Done.  I did that originally, but it seemed like repetitive code to me.  Also, the pruning unit tests did pass with the logic at the top of the function.  However, I will defer to you on this and the pruning is quite complex and I don't want to break it.